### PR TITLE
Report all probes

### LINKED ISSUE #45: Add ability to name & log an arbitrary number of temperature probes
I have additional temp probes that aren't one of the beer/fridge/room options. I'm using fermentrack/brewpi-8266 to control glycol chill loops on my fermenters.  I'm ok with the control probes not have "correct" names (for example, my fermenter probe is configured as the fridge probe in fermentrack), but I'd like to at least be able to log the data from these extra probes.  

I personally have a probe for the coolant bath and for the post-jacket coolant temp, but I think any solution should allow arbitrary names to be set on these non-control probes.

### PR DESCRIPTION
**Pull request by speshak**
Fixes #45

### CODE REVIEW

**Comment by speshak** [comment]
I'm opening this now as more of an RFC before I go further.  There are a couple outstanding issues that at least need discussed:

- The "keywords" I chose for extending the PiLink protocol.  I used `T` and `o`, but there might be better options.
- Being more defensive in how the custom names are stored.  If you were being malicious, you could overwrite some of the other system data by claiming a device address of `/mdns.txt`
- Is there a reason why `parseJsonToken` [eats spaces](https://github.com/thorrak/brewpi-esp8266/blob/esp32/src/PiLink.cpp#L763)?  This means that no probe name can have a space.  That could be a reasonable trade off, especially if other code paths depend on that behavior, but it was a bit surprising when I was testing.

**Comment by speshak** [comment]
I've answered my own question about `parseJsonToken`.  That function is used for both JSON keys and values, and spaces (and the other skipped chars) aren't valid in a key.

**Comment by speshak** [comment]
I fixed the issue with possibly squashing other SPIFFS content, and added another PiLink command (`p`) for dumping the list of configured device names.  I think this is ready for review.

File: src/DeviceNameManager.cpp [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+ * Set a human readable name for a device.
+ * @param device - The identifier for the device, most commonly the OneWire device address (in hex)
+ * @param name - The name to set
+ */
+void DeviceNameManager::setDeviceName(const char* device, const char* name)
+{
+  char filename[32];
+  DeviceNameManager::deviceNameFilename(filename, device);
+
+  File f = SPIFFS.open(filename, "w");
```
Comment (thorrak) [inline src/DeviceNameManager.cpp:39]: At some point (later!) we should probably revisit how the device definitions are written out and see if they can also be refactored into JSON writes. Project for later!

**Pull request merged**